src/main/java/com/sucy/skill/dynamic/trigger/ProjectileHitTrigger.java

```diff
@@ -0,0 +1,71 @@
+package com.sucy.skill.dynamic.trigger;
+
+import com.sucy.skill.api.Settings;
+import com.sucy.skill.api.particle.target.EntityTarget;
+import com.sucy.skill.dynamic.TempEntity;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.entity.ProjectileHitEvent;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * ProSkillAPI © 2023
+ * com.sucy.skill.dynamic.trigger.ProjectileHitTrigger
+ */
+public class ProjectileHitTrigger implements Trigger<ProjectileHitEvent> {
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public String getKey() {
+        return "PROJECTILE_HIT";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Class<ProjectileHitEvent> getEvent() {
+        return ProjectileHitEvent.class;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean shouldTrigger(ProjectileHitEvent event, int level, Settings settings) {
+        List<String> projectiles = settings.getStringList("projectile");
+        String type = settings.getString("type","both");
```

Comment:
It would probably be good to add `block` as an option here too. So we'd consider it hit on either block, entity, or both depending on the setting.

Reply:
Here actually if the `type` is anything but `both` or `entity`, it will be considered as a `block`. Should i add validation so `type` would accept only `entity`, `block` or `both` values?